src/InstallWizard.js

```diff
@@ -152,12 +152,18 @@ class InstallWizard extends Component {
     // and other items related to the update progress in the progress.json.
     // If both steps and currentMenuName are present, the update progress
     // is on going, otherwise, no progress.
-    if(responseData && responseData.currentMenuName && responseData.steps) {
-      let steps = responseData.steps;
-      let pages =
-        this.getPages(steps);
+    if(responseData && responseData.currentMenuName) {
       let updateStates = responseData;
-      updateStates.pages = pages;
+      if(responseData.steps) {
+        let steps = responseData.steps;
+        let pages =
+          this.getPages(steps);
+        updateStates.pages = pages;
+      }
+      else {
+        // if there is no progress, will record the menuName passed in
+        updateStates.currentMenuName = this.props.menuName;
```

Comment:
this if block confuses me a little... its checking for steps, but sets currentMenuName... I assume thats intentional, but its a bit confusing

Reply:
@jeremy-moffitt  will try to update the comments

Reply:
ok, not sure update will talk about it first.